deepchecks/recommender/checks/date_train_test_leakage_overlap.py

```diff
@@ -0,0 +1,171 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""The date_leakage check module."""
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.tabular import TrainTestCheck
+from deepchecks.recommender import Context
+from deepchecks.utils.strings import format_datetime, format_percent
+from deepchecks.tabular.utils.task_type import TaskType
+import plotly.graph_objects as go
+__all__ = ['DateTrainTestLeakageOverlap']
+
+
+class DateTrainTestLeakageOverlap(TrainTestCheck):
+    """Check test data that is dated earlier than the latest date in train. \
```

Comment:
This docstring needs some re-writing. From here it sounds like this check is *doing* the cross val for the user, while in reality the user, that was using cross val, should set validation_per_user=True so that this check won't alert on irrelevant types of leakage for his usecase.